[JSC] Allow LocalOverrides.xcconfig to disable VerifyModule/TAPI checks

kmiller68 opened the pull request:
#### ed385081598ca53921e3152b693c39a694c84f73
<pre>
[JSC] Allow LocalOverrides.xcconfig to disable VerifyModule/TAPI checks
<a href="https://bugs.webkit.org/show_bug.cgi?id=280276">https://bugs.webkit.org/show_bug.cgi?id=280276</a>
<a href="https://rdar.apple.com/136597730">rdar://136597730</a>

Reviewed by Elliott Williams.

These validation steps take upwards of 1 minute for incremental builds. I&apos;ve yet to have either find a bug for me so
this patch adds settings to disable them: `JSC_OVERRIDE_ENABLE_MODULE_VERIFIER` and `JSC_OVERRIDE_SUPPORTS_TEXT_BASED_API`
If these are not set the current settings are used.

* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:

Canonical link: <a href="https://commits.webkit.org/284179@main">https://commits.webkit.org/284179@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/106113c1c488f44c64f85789d5acb0506f5d997b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68657 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48049 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21316 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/72726 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19802 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55845 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/19618 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/54747 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13163 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71724 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43913 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59281 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35205 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40579 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/16705 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18159 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/61775 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62531 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17052 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/74420 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/67905 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12628 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16305 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/62229 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12668 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59360 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62256 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15244 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10204 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3815 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/89684 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43850 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15885 "Found 10 new JSC stress test failures: wasm.yaml/wasm/function-tests/many-arguments-to-function.js.default-wasm, wasm.yaml/wasm/function-tests/many-arguments-to-function.js.wasm-bbq, wasm.yaml/wasm/fuzz/memory.js.default-wasm, wasm.yaml/wasm/stress/js-wasm-call-many-return-types-on-stack-no-args.js.default-wasm, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.default-wasm, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-bbq, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-collect-continuously, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-slow-memory, wasm.yaml/wasm/stress/tail-call.js.wasm-bbq, wasm.yaml/wasm/stress/tail-call.js.wasm-eager-jettison (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/44924 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46118 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44666 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
```diff
@@ -48,7 +51,10 @@ WK_SUPPORTS_OTHER_TAPI_FLAGS_STATICLIBS_WATCHOS_SINCE_10 = YES;
 WK_SUPPORTS_OTHER_TAPI_FLAGS_STATICLIBS_xros = YES;
 WK_SUPPORTS_OTHER_TAPI_FLAGS_STATICLIBS_xrsimulator = YES;
 
-SUPPORTS_TEXT_BASED_API = $(WK_AND_$(TAPI_USE_SRCROOT)_$(WK_SUPPORTS_OTHER_TAPI_FLAGS_STATICLIBS));
+SUPPORTS_TEXT_BASED_API = $(SUPPORTS_TEXT_BASED_API_$(JSC_OVERRIDE_SUPPORTS_TEXT_BASED_API))
+SUPPORTS_TEXT_BASED_API_ = $(WK_AND_$(TAPI_USE_SRCROOT)_$(WK_SUPPORTS_OTHER_TAPI_FLAGS_STATICLIBS));
+SUPPORTS_TEXT_BASED_API_NO = NO;
+SUPPORTS_TEXT_BASED_API_YES = YES;
```
There is also a TAPI_ENABLE_VERIFICATION_MODE setting, could you see if that works to remove or speed up the generation phase? I think we should continue declaring TAPI support and only tell it to skip the verification step. We had some problems with incremental builds when toggling this setting.

emw-apple reviewed (approved):
r+ with one suggested change.

kmiller68 committed:
[JSC] Allow LocalOverrides.xcconfig to disable VerifyModule/TAPI checks
https://bugs.webkit.org/show_bug.cgi?id=280276
rdar://136597730

Reviewed by Elliott Williams.

These validation steps take upwards of 1 minute for incremental builds. I've yet to have either find a bug for me so
this patch adds settings to disable them: `JSC_OVERRIDE_ENABLE_MODULE_VERIFIER` and `JSC_OVERRIDE_SUPPORTS_TEXT_BASED_API`
If these are not set the current settings are used.

* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:

Canonical link: https://commits.webkit.org/284179@main
